vmdb/app/controllers/catalog_controller.rb

```diff
@@ -1222,13 +1222,17 @@ def set_resource_action(st)
     ]
     actions.each do |action|
       ra = st.resource_actions.find_by_action(action[:name])
-      unless ra
+      if ra.nil? && !@edit[:new][action[:edit_key]].blank?
         attrs = {:action => action[:name]}
         ra = st.resource_actions.build(attrs)
       end
-      ra.dialog = d
-      ra.fqname = @edit[:new][action[:edit_key]] unless @edit[:new][action[:edit_key]].nil?
-      ra.save!
+      if @edit[:new][action[:edit_key]].blank?
+        st.resource_actions.where(:action => action[:name]).first.try(:destroy)
+      else
+        ra.dialog = d
+        ra.fqname = @edit[:new][action[:edit_key]].blank? ? nil : @edit[:new][action[:edit_key]]
+        ra.save!
```

Comment:
You do not need the check for `@edit[:new][action[:edit_key]]` being blank since that case is handled by the `if` statement.
The then entire else section could be reduced to:
`ra.update_attributes(:dialog => d, :fqname => @edit[:new][action[:edit_key]])`